src/test/java/uk/gov/hmcts/reform/sendletter/services/StaleLetterServiceTest.java

```diff
@@ -0,0 +1,183 @@
+package uk.gov.hmcts.reform.sendletter.services;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.sendletter.entity.Letter;
+import uk.gov.hmcts.reform.sendletter.entity.LetterRepository;
+import uk.gov.hmcts.reform.sendletter.entity.LetterStatus;
+import uk.gov.hmcts.reform.sendletter.model.out.StaleLetter;
+import uk.gov.hmcts.reform.sendletter.services.date.DateCalculator;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+import static java.time.ZoneOffset.UTC;
+import static java.time.temporal.ChronoUnit.DAYS;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.sendletter.util.TimeZones.EUROPE_LONDON;
+
+@ExtendWith(MockitoExtension.class)
+public class StaleLetterServiceTest {
+
+    @Mock
+    private DateCalculator dateCalculator;
+
+    @Mock
+    private LetterRepository letterRepository;
+
+    @Mock
+    private Clock clock;
+
+    @BeforeEach
+    public void setUp() {
+        given(letterRepository.findStaleLetters(any())).willReturn(Stream.of());
+        given(dateCalculator.subtractBusinessDays(any(), anyInt())).willReturn(ZonedDateTime.now());
+
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of(EUROPE_LONDON));
+        setCurrentTimeAndZone(EUROPE_LONDON, now);
+    }
+
+    @Test
+    public void getStaleLetters_should_call_date_calculator_with_current_time_when_before_ftp_downtime_start() {
+        // given
+        String ftpDowntimeStartTime = "16:00";
+        int minStaleLetterAgeInBusinessDays = 123;
+
+        ZonedDateTime now = LocalDate.parse("2019-05-03")
+            .atTime(15, 59) // currently it's before FTP downtime
+            .atZone((ZoneId.of(EUROPE_LONDON)));
+
+        setCurrentTimeAndZone(EUROPE_LONDON, now);
+
+        // when
+        staleLetterService(ftpDowntimeStartTime, minStaleLetterAgeInBusinessDays).getStaleLetters();
```

Comment:
missing stream closure

Reply:
This is a 100% in-memory stream, that lives in the scope of a single test. There's no need to close it, as it's guaranteed to be GC-collected. It's a similar situation to why you don't have to explicitly close a stream retrieved from an ArrayList.

It's crucial to close streams behind which there are unmanaged resources (this is the whole point of AutoClosable). For example, when consuming a stream retrieved from a JPA repository, we have to make sure it's always closed at the end. It's not the case here, though.